### DISCUSSION TITLE
[FEAT] Allow Players to Remove Growing Crops (Shovel) (Discussion...?)

### DISCUSSION THREAD

**Discussion by ShiningMinions** [sunflower-land/sunflower-land#1126] [Ideas]
## This feature request got migrated from the following Github Issue:
https://github.com/sunflower-land/sunflower-land/issues/1116

## Pixelized Crib Trello Ref:
https://trello.com/c/pARkxAiI

## Summary of Request

As a player, I want a way to remove crops from my field (which are currently growing), so that I can correct seeds I mistakenly planted, and I don't have to wait the entire grow cycle to re-plant.

## Proof of Community Demand

- [Aha! Proposal #107](https://sunflower-land.ideas.aha.io/ideas/SFL-I-107)
- [Aha! Proposal #68](https://sunflower-land.ideas.aha.io/ideas/SFL-I-68)
- VIP Channel Poll:
![VIP Poll](https://trello.com/1/cards/629e0275b4a9622820eb2aa6/attachments/62a6949b7b8f570653028139/previews/62a6949b7b8f570653028155/download/community-demand-remove-crops-vip.png)
- HODL Channel Poll:
![HODL Poll](https://trello.com/1/cards/629e0275b4a9622820eb2aa6/attachments/62a6949a7902252256c2b27a/previews/62a6949a7902252256c2b28d/download/community-demand-remove-crops-hodl.png)

## Contributors

If know, please provide the list of contributors who will be working on this feature.

- TBD

## README - Pre-Notes:

We believe every farm should have the ability to remove unwanted crops by default - it's very frustrating as a player not being able to remove crops, and we feel every player should have this ability by default. However, we do like the idea of minting "better shovels" at some point in the future. For example, maybe a "wooden shovel" will give you 10% chance to recover seed. Maybe "stone shovel" would be 25%, etc. Inspiration for this idea came from [Aha! Proposal 107](https://sunflower-land.ideas.aha.io/ideas/SFL-I-107).

As a quick future brainstorm, we think the ability to mint better shovels could be unlocked only after completing some new "quest". This would be extremely fun. The quest lore could have something to do with the goblin who steals your shovel. He could "teach you the secret of crafting better shovels", or something like this. This is out-of-scope for this feature request, but could be extremely fun content for future.

## Technical Requirements

1. All farms should be able to remove crops. Therefore, every farm should start with exactly one "rusty shovel" by default. This "rusty shovel" is a non-withdrawable inventory item, and its only function is to remove unwanted crops currently growing in the player's plots.
 a. Description: "An old rusty shovel. Equip to start removing unwanted crops."
 b. The "rusty shovel" should automatically be added to every existing farm's inventory.
 c. The "rusty shovel" should exist in inventory the moment you mint your farm.

2. When you click on a crop **with your "rusty shovel" equipped**:
  a. If the crop is ready-to-harvest, then skip the rest of this logic. Simply harvest the crop like normal.
  b. Otherwise, for the **first crop you attempt to remove**, display a warning modal:
   1. Modal Title: Remove Growing Crop?
   2. Modal Body: `Warning! You have your shovel equipped. You will not recover the crop or it's seed. Are you sure you want to start removing crops?`
   3. Display two buttons: `Yes, Start Removing` and `Cancel`.
   4. Clicking `Yes, Start Removing` removes the first crop.

3. After clicking `Yes, Start Removing` button, the player can start removing crops at will. However, if one of the following conditions occurs, then the Warning Modal should be triggered to display again:
  a. Player switches inventory items, or
  b. 10 seconds of time pass with no crops being removed.

4. Player **should not** be allowed to remove crops if the goblin has stolen their shovel. Player will need to recover shovel first.

### Backend Updates Needed

1. Support crop removals from the player's gameState.
2. Provide a "rusty shovel" to existing farms.
3. Include a "rusty shovel" for newly minted farms.

### Smart Contract Updates Needed

1. Maybe, don't make the "rusty shovel" item an NFT, as it will be non-withdraw-able and every farm will have exactly 1. But if that's not possible, then potentially will need to add this as a new NFT item to a smart contract.

## UI/UX Design Requirements

N/A

## Assets Needed

- A "rusty shovel" asset (credit @jestereinsteinibasan - LittleEinstein):

![Rusty Shovel](https://trello.com/1/cards/629e0275b4a9622820eb2aa6/attachments/62b32c22862a948b6113a8d2/previews/62b32c23862a948b6113a8f2/download/rusty_shovel.png)

![Rusty Shovel 3x](https://trello.com/1/cards/629e0275b4a9622820eb2aa6/attachments/62b32c238f80e71c6f348fe1/previews/62b32c248f80e71c6f34900d/download/rusty_shovel_3x.png)

**Comment by ShiningMinions** [comment]
@spencerdezartsmith I'm confused why this was asked to be moved to a Github discussion - I made these templates and they got accepted-in. Are these not going to be used anymore? I used the "Feature Request" template for this:

![github_templates](https://user-images.githubusercontent.com/103600068/175182202-3536615a-ec7c-429e-881a-050636ca9923.png)

**Reply by Yougottabekiddingme123** [reply]
Yip. Definitely  need this. Everyone makes a mistake once in a while plating the wrong crop. We should be able to remove it amd replant.

**Comment by ShiningMinions** [comment]
This discussion should be closed in favor of re-opened issue: https://github.com/sunflower-land/sunflower-land/issues/1116
